packages/core/src/components/Dialog/Dialog.jsx

```diff
@@ -0,0 +1,107 @@
+import AriaModal from 'react-aria-modal';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export const Dialog = function(props) {
+  const {
+    actions,
+    children,
+    escapeExitDisabled,
+    onExit,
+    title,
+    ...modalProps
+  } = props;
+
+  return (
+    <AriaModal
+      dialogClass="ds-c-dialog"
+      escapeExits={!escapeExitDisabled}
+      includeDefaultStyles={false}
+      onExit={onExit}
+      titleId="dialog-title"
+      underlayClass="ds-c-dialog-wrap"
+      {...modalProps}
+    >
+      <header className="ds-c-dialog__header">
+        {title && (
+          <h1 className="ds-h2" id="dialog-title">
+            {title}
+          </h1>
+        )}
+        <button
+          className="ds-c-button ds-c-button--transparent ds-c-dialog__close"
```

Comment:
What do you think about adding an optional `aria-label` so we can provide a richer label to screen readers for the action buttons in the modal. I'd advise an aria-label for at least the Close button, and the Cancel button, because those could effectively say the same thing with one prop, unless they are intended to do different things.

I feel the extra context is good for assistive devices because if the amount of information being read is long, or time has lapsed since the user opened the modal, they may not remember what they are closing.

Reply:
Good idea 👍